src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -419,7 +419,7 @@ private void CheckUnarySignature(DiagnosticBag diagnostics)
             // SPEC: A unary + - ! ~ operator must take a single parameter of type
             // SPEC: T or T? and can return any type.
 
-            if (!this.GetParameterType(0).StrippedType().TupleUnderlyingTypeOrSelf().Equals(this.ContainingType, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
+            if (!this.GetParameterType(0).StrippedType().Equals(this.ContainingType, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
```

Comment:
>Equals [](start = 57, length = 6)

Should this `Equals()` and the one below include `TypeCompareKind.IgnoreTupleNames`?

If so, please consider extracting a helper method for all of these checks.